python/ray/train/v2/tests/test_controller.py

```diff
@@ -253,21 +280,31 @@ def test_worker_group_start_failure(error_type):
     worker_group: DummyWorkerGroup = controller.get_worker_group()
     worker_group.set_start_failure(error_type)
 
+    assert isinstance(controller.get_state(), InitializingState)
+
     scaling_policy.queue_recovery_decision(
         ResizeDecision(num_workers=2, resources_per_worker={})
     )
+
+    controller._run_control_loop_iteration()
+    assert isinstance(controller.get_state(), SchedulingState)
+
     # Worker group will fail to start, but controller should not raise
-    # and should go into RECOVERING state.
+    # and should go into RESCHEDULING state.
     controller._run_control_loop_iteration()
-    assert controller.get_state() == TrainControllerState.RECOVERING
+    assert isinstance(controller.get_state(), ReschedulingState)
```

Comment:
nice! forgot we had this test

Reply:
These existing tests were very nice and easy to modify to test the behavior of this PR. Thanks to whoever wrote them 😃